site/app/libraries/database/DatabaseQueries.php

```diff
@@ -636,6 +636,19 @@ public function createPost($user, $content, $thread_id, $anonymous, $type, $firs
         return $this->course_db->rows()[0]["max_id"];
     }
 
+    public function findPost($thread_id) {
+        $this->course_db->query("SELECT * from posts where thread_id = ?", [$thread_id]);
+        return $this->course_db->row();
+    }
+
+    public function existsAnnouncementsId($thread_id) {
+        $this->course_db->query("SELECT title from threads where id = ?", [$thread_id]);
+        $title = $this->course_db->row()['title'];
+        $announcement_subject = "[Submitty " . $this->core->getConfig()->getCourse() . "]: New Announcement: (" . $thread_id . ") " . $title;
+        $this->submitty_db->query("SELECT 1 from emails where subject = ?", [$announcement_subject]);
```

Comment:
What is the purpose of this query?

Reply:
This was the way we came up with to check if the announcement was made, since there is no easy and clean way to check if the announcement was made at the moment without changing the database. We decided since the feature isn't going to be used too much, we should do this instead which was easier than making a migration.